controllers/rclone_mover_test.go

```diff
@@ -0,0 +1,671 @@
+						}
+						pvc = &corev1.PersistentVolumeClaim{}
+						Eventually(func() error {
+							return k8sClient.Get(ctx, types.NamespacedName{Name: pvcName, Namespace: rd.Namespace}, pvc)
+						}, maxWait, interval).Should(Succeed())
+						Expect(*pvc.Spec.StorageClassName).To(Equal(scName))
+					})
+				})
+			})
+
+			When("Using a copy method of Snapshot", func() {
+				BeforeEach(func() {
+					rd.Spec.Rclone.CopyMethod = scribev1alpha1.CopyMethodSnapshot
+				})
+
+				It("Ensure that a VolumeSnapshot is created at the end of an iteration", func() {
+					// get list of volume snapshots in the ns
+					snapshots := &snapv1.VolumeSnapshotList{}
+					Eventually(func() []snapv1.VolumeSnapshot {
+						_ = k8sClient.List(ctx, snapshots, client.InNamespace(rd.Namespace))
+						return snapshots.Items
+					}, maxWait, interval).Should(Not(BeEmpty()))
+					// update the VS name
+					snapshot := snapshots.Items[0]
+					foo := "dummysnapshot"
+					snapshot.Status = &snapv1.VolumeSnapshotStatus{
+						BoundVolumeSnapshotContentName: &foo,
+					}
+					Expect(k8sClient.Status().Update(ctx, &snapshot)).To(Succeed())
+					// wait for an image to be set for RD
+					Eventually(func() *corev1.TypedLocalObjectReference {
+						_ = k8sClient.Get(ctx, utils.NameFor(rd), rd)
+						return rd.Status.LatestImage
+					}, maxWait, interval).Should(Not(BeNil()))
+					latestImage := rd.Status.LatestImage
+					// ensure the name was correctly set
+					Expect(latestImage.Kind).To(Equal("VolumeSnapshot"))
+					Expect(*latestImage.APIGroup).To(Equal(snapv1.SchemeGroupVersion.Group))
+					Expect(latestImage).To(Not(Equal("")))
+				})
+
+				When("When a VolumeSnapshotClass is specified", func() {
+					vscName := "MyVolumeSnapshotClass"
+					BeforeEach(func() {
+						rd.Spec.Rclone.ReplicationDestinationVolumeOptions.VolumeSnapshotClassName = &vscName
+					})
+
+					It("is used as the VSC for the Snapshot", func() {
+						// create a snapshot & verify the VSC matches
+						snapshots := &snapv1.VolumeSnapshotList{}
+						Eventually(func() []snapv1.VolumeSnapshot {
+							_ = k8sClient.List(ctx, snapshots, client.InNamespace(rd.Namespace))
+							return snapshots.Items
+						}, maxWait, interval).Should(Not(BeEmpty()))
+						snapshot := snapshots.Items[0]
+						Expect(*snapshot.Spec.VolumeSnapshotClassName).To(Equal(vscName))
+					})
+				})
+			})
+
+			It("Job set to fail", func() {
+				Eventually(func() error {
+					return k8sClient.Get(ctx, utils.NameFor(job), job)
+				}, maxWait, interval).Should(Succeed())
+				// force job fail state
+				Eventually(func() error {
+					// make sure job object is consistent
+					if err := k8sClient.Get(ctx, utils.NameFor(job), job); err != nil {
+						return err
+					}
+					job.Status.Failed = *job.Spec.BackoffLimit + 1000
+					return k8sClient.Status().Update(ctx, job)
+				}, maxWait, interval).Should(Succeed())
+				// job should eventually be restarted
+				Eventually(func() error {
```

Comment:
I think you need to check that the job is no longer failed (i.e., it's been deleted and re-created by the controller). As is, I think this will pass w/ the existing failed job.

Reply:
Updated this, also did what we discussed earlier and moved the test out of the block marking the job as successful.